File: lab2no2.swift
func illustration(banki: Int, ostatok: Int, count_steps: Int) {
    if count_steps == 0 {
        print("Jars")
        print(String(repeating: "F", count: banki))
    } else if count_steps % 2 == 1 {
        print("\(count_steps) step")
        print(String(repeating: "E", count: banki))
    } else {
        print("\(count_steps) step")
        print(String(repeating: "F", count: banki - ostatok))
        print(String(repeating: "E", count: ostatok))
    }
}

print("Input jar quantity:")
if let banki = Int(readLine() ?? ""), banki > 0 {
    print("Input exchange:")
    if let obmen = Int(readLine() ?? ""), obmen > 0 {
        var count_empty = 0
        var count_steps = 0
        var ostatok = 0
        var new_banki = banki

        while new_banki >= obmen {
            if count_steps == 0 {
                illustration(banki: new_banki, ostatok: ostatok, count_steps: count_steps)
            }
            count_steps += 1
            illustration(banki: new_banki, ostatok: ostatok, count_steps: count_steps)
            count_empty += new_banki - ostatok
            ostatok = new_banki % obmen
            new_banki = (new_banki / obmen) + ostatok
            count_steps += 1
            illustration(banki: new_banki, ostatok: ostatok, count_steps: count_steps)
        }

        if new_banki < obmen {
            count_empty += new_banki - ostatok
            count_steps += 1
            illustration(banki: new_banki, ostatok: ostatok, count_steps: count_steps)
            print("Jars drunk: \(count_empty)")
            print("Step count: \(count_steps)")
        }
    } else {
        print("Error")
    }
} else {
    print("Error")
}

File: lab2no2.rs
use std::io;

fn illustration(banki: i32, ostatok: i32, count_steps: i32) {
    if count_steps == 0 {
        println!("Jars");
        for _ in 0..banki {
            print!("F");
        }
        println!();
    } else if count_steps % 2 != 0 {
        println!("{} step", count_steps);
        for _ in 0..banki {
            print!("E");
        }
        println!();
    } else {
        println!("{} step", count_steps);
        for _ in 0..(banki - ostatok) {
            print!("F");
        }
        println!();
        for _ in 0..ostatok {
            print!("E");
        }
        println!();
    }
}

fn main() {
    let mut input = String::new();

    println!("Input jar quantity: ");
    io::stdin().read_line(&mut input).unwrap();
    let banki: i32 = input.trim().parse().unwrap();
    input.clear();

    println!("Input exchange: ");
    io::stdin().read_line(&mut input).unwrap();
    let obmen: i32 = input.trim().parse().unwrap();

    if banki <= 0 || obmen <= 0 {
        println!("Error");
    } else {
        let mut count_empty = 0;
        let mut count_steps = 0;
        let mut ostatok = 0;
        let mut new_banki = banki;

        while new_banki >= obmen {
            if count_steps == 0 {
                illustration(new_banki, ostatok, count_steps);
            }
            count_steps += 1;
            illustration(new_banki, ostatok, count_steps);
            count_empty += new_banki - ostatok;
            ostatok = new_banki % obmen;
            new_banki = (new_banki / obmen) + ostatok;
            count_steps += 1;
            illustration(new_banki, ostatok, count_steps);
        }

        if new_banki < obmen {
            count_empty += new_banki - ostatok;
            count_steps += 1;
            illustration(new_banki, ostatok, count_steps);
            println!("Jars drunk: {}", count_empty);
            println!("Step count: {}", count_steps);
        }
    }
}
